src/main/java/org/elasticsearch/discovery/srv/SrvUnicastHostsProvider.java

```diff
@@ -122,6 +122,13 @@ protected Resolver buildResolver(Settings settings) {
                 logger.warn("Could not create resolver. Using default resolver.", e);
             }
         }
+
+        parent_resolver = parent_resolver == null ? Lookup.getDefaultResolver() : parent_resolver;
```

Comment:
I guess my (entirely theoretical) concern is that setting TCP on the default resolver might affect other plugins that also use dnsjava and the default resolver. It's a remote possibility, so I'm not suggesting we put effort into supporting that case, just documenting it here in case it comes up again.